dawn/src/dawn/Compiler/DawnCompiler.cpp

```diff
@@ -168,7 +171,12 @@ DawnCompiler::lowerToIIR(const std::shared_ptr<SIR>& stencilIR) {
   optimizer.pushBackPass<PassTemporaryType>();
   optimizer.pushBackPass<PassLocalVarType>();
   optimizer.pushBackPass<PassRemoveScalars>();
-  optimizer.pushBackPass<PassStageSplitter>();
+  if(stencilIR->GridType == ast::GridType::Unstructured) {
```

Comment:
I know you mention in the description that you don't want to be smart about the splitting in the lowering and I agree very much. 
I have an implementation of the parallel model coming that does exactly this (1 stage per stmt) at the lowering level. But for now I would say that `PassStageSplitAllStatements` would be beneficial for both models as it is a step towards implementing our execution model. (MS loop is still to be done). So I would remove the if here and use it for both

Reply:
Perfect, we didn't dare to touch it, but since you will anyway do it, we can do it already here. However, if performance changes, I think it's not in the scope of this PR to fix it.

Reply:
I would say, we merge this PR with this if, then I'll open a new PR, where it is removed and we can investigate if things break. Ok?

Reply:
2 tests in dawn fail actually, so I will not remove the if in this PR.

Reply:
sounds ok to me